znai-reactjs/src/doc-elements/Documentation.js

```diff
@@ -487,24 +487,23 @@ export class Documentation extends Component {
     //
     onPageUpdate(pageProps) {
         const updatePagesReference = () => this.getAllPagesPromise().then((allPages) => {
-            this.updatePagesReference(allPages, pageProps);
+            allPages.update(pageProps)
         })
 
         this.navigateToPageAndDisplayChange(pageProps, updatePagesReference)
     }
 
     // one of the files referred from a markup or multiple markups was changed
-    // we need to update multiple pages at once and either refresh current view (if one of the changed pages is it)
+    // we need to update multiple pages at once and either refresh current view (if one of the changed pages is current page)
     // or navigate to the first modified page
     //
     onMultiplePagesUpdate(listOfPageProps) {
         const updatePagesReference = () => this.getAllPagesPromise().then((allPages) => {
```

Comment:
Can you move this local method to class method, so you can reuse it within `onPageUpdate`, too?